server/transport.go

```diff
@@ -85,6 +85,11 @@ func decodeSCEPRequest(ctx context.Context, r *http.Request) (interface{}, error
 	}
 	defer r.Body.Close()
 
+	operation := r.URL.Query().Get("operation")
+	if len(operation) == 0 {
+		return nil, &BadRequestError{Message: "missing operation"}
+	}
```

Comment:
Might be worth double-checking by @roperzh who's more familiar than I am with the macOS MDM flow - my concern is that this could generate errors from requests that currently succeeded (but presumably ignored/no-ops) if there was no operation?

Reply:
to be honest I had to go and look at [the spec](https://datatracker.ietf.org/doc/html/rfc8894#name-http-post-and-get-message-f) because I didn't know the answer to this.

from what I can see, some messages can have empty operations so +1 to remove this.

Reply:
Gotcha; fwiw, currently if you leave out the operation, `fleet` responds with a 500. Removing this code would revert back to that behavior. Should I instead update the code to allow empty operations? Ditto for an empty message (like [here](https://github.com/fleetdm/scep/pull/1#discussion_r1447919363))

Reply:
> currently if you leave out the operation, fleet responds with a 500.

Ah gotcha, yeah if it did generate an error previously, then the added validation makes sense to me. I thought this case might've been a successful no-op before.

Reply:
I think you're right about `operation`, taking a second look here:

https://github.com/fleetdm/scep/blob/3aa7a5a9d937a028c081e46f605c4a85fbcdadde/server/endpoint.go#L137-L149

just as a curiosity, I also see this:

> Note that when used with HTTP POST, the only OPERATION possible is "PKIOperation", so many CAs don't check this value or even notice its absence